src/strings/ops.c

```diff
@@ -337,6 +389,15 @@ static void iterate_gi_into_string(MVMThreadContext *tc, MVMGraphemeIter *gi, MV
             MVM_string_gi_next_strand_rep(tc, gi);
         }
     }
+    if (result->body.num_graphs <= 2 && result->body.storage_type == MVM_STRING_GRAPHEME_32) {
+        MVMuint8 i;
+        MVMGrapheme32 *old = result->body.storage.blob_32;
+        for (i = 0; i < result->body.num_graphs; i++) {
+            result->body.storage.in_situ_32[i] = old[i];
+        }
+        result->body.storage_type    = MVM_STRING_IN_SITU_32;
+        MVM_free(old);
```

Comment:
it would be interesting to see how often this triggers; should we be able to know `result->body.num_graphs` near the start of the function, and could we then always never go with storage type string_grapheme_32 throughout the body of the function?

Reply:
You're right, we could probably do it right around https://github.com/MoarVM/MoarVM/pull/1802/files#diff-bc76be5614df2473912916db269fc246c79efbc83a0a065ee8b785aa3c68d040R330-R333